### PR TITLE
FsReveal style auto refresh

### PR DESCRIPTION
**Pull request by robertpi**
- Use suave as web server instead of custom server
- copy and adapt code from RsReveal to detect changes
- Add javascript need to listen for changes

### CODE REVIEW

**Comment by cdrnet** [comment]
Great, thanks!

Should we maybe add a paket framework constraint in the dependencies file to avoid vnext dependencies leaking in?

**Comment by robertpi** [comment]
I guess. 

I had to remove the constrain on FSharp.Formatting to ensure we got the right version of RazorEngine. With constraint I kept getting an assembly load exception as conflicting versions of RazorEngine were needed. Maybe there's a better way to solve it?

File: paket.dependencies [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
 
 nuget FSharp.Configuration
-nuget FSharp.Formatting !~> 2.4.0
+nuget FSharp.Formatting
 nuget FsUnit.xUnit
 nuget xunit.runners
 nuget NuGet.CommandLine
 nuget FAKE
+nuget RazorEngine
+nuget suave
```
Comment (forki) [inline paket.dependencies]: :heart:

File: paket.lock [hunk 2]
```diff
@@ -1,20 +1,145 @@
...
+      FSharp.Core (>= 3.1.2.5)
+      xunit (2.1.0)
+    Microsoft.AspNet.Razor (3.2.3) - framework: >= net45
+    NuGet.CommandLine (2.8.6)
+    RazorEngine (3.7.3)
+      Microsoft.AspNet.Razor (>= 3.0.0) - framework: >= net45
+      Microsoft.AspNet.Razor (2.0.30506.0) - framework: net40
+    Suave (0.33.0)
+      FSharp.Core (>= 3.1.2.5)
+    System.Collections (4.0.10) - framework: dnxcore50
```
Comment (forki) [inline paket.lock]: this dnxcore50 crap is hilarious

**Comment by cdrnet** [comment]
Merged anyway, assuming it does not cause any problems (but seemingly does fix an issue around RazorEngine).

**Pull request merged**